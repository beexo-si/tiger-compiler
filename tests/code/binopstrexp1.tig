/*TIGER_ARGS=-canon -code*/
let
    var s:="asd"
    var x:="asd"
in
    x=s
end
