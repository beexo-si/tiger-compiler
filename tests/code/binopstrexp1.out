bien!
------Canon------
L2:	.long 3
	.ascii "asd"
L1:	.long 3
	.ascii "asd"
;;--END-STRS--:
--FRAME L0:
LABEL L6
MOVE(
 TEMP T4,
 TEMP rdi)
MOVE(
 TEMP T5,
 TEMP rsi)
MOVE(
 TEMP T6,
 TEMP rbx)
MOVE(
 TEMP T7,
 TEMP r12)
MOVE(
 TEMP T8,
 TEMP r13)
MOVE(
 TEMP T9,
 TEMP r14)
MOVE(
 TEMP T10,
 TEMP r15)
MOVE(
 MEM(
  BINOP(PLUS,
   TEMP rbp,
   CONST -8)),
 TEMP rcx)
MOVE(
 TEMP T0,
 NAME L1)
MOVE(
 TEMP T1,
 NAME L2)
MOVE(
 TEMP T3,
 CONST 1)
EXP(
 CALL(
  NAME _stringCompare,
   TEMP T1,
   TEMP T0))
MOVE(
 TEMP T2,
 TEMP rax)
CJUMP(EQ,
 TEMP T2,
 CONST 0,
 L3,L4)
LABEL L4
MOVE(
 TEMP T3,
 CONST 0)
LABEL L3
MOVE(
 TEMP rax,
 TEMP T3)
MOVE(
 TEMP rdi,
 TEMP T4)
MOVE(
 TEMP rsi,
 TEMP T5)
MOVE(
 TEMP rbx,
 TEMP T6)
MOVE(
 TEMP r12,
 TEMP T7)
MOVE(
 TEMP r13,
 TEMP T8)
MOVE(
 TEMP r14,
 TEMP T9)
MOVE(
 TEMP r15,
 TEMP T10)
JUMP( NAME L5)
LABEL L5
;;-END-FRAME-:
;;--FRAME--L0:
LABEL: L6
MOVE: movq %'s0, %'d0 D:T4 S:rdi
MOVE: movq %'s0, %'d0 D:T5 S:rsi
MOVE: movq %'s0, %'d0 D:T6 S:rbx
MOVE: movq %'s0, %'d0 D:T7 S:r12
MOVE: movq %'s0, %'d0 D:T8 S:r13
MOVE: movq %'s0, %'d0 D:T9 S:r14
MOVE: movq %'s0, %'d0 D:T10 S:r15
MOVE: movq %'s0, %'d0 D:T11 S:rbp
OPER: addq %'d0, $-8 D:[T11] S:[T11]
OPER: movq %'s0, (%'s1) D:[] S:[rcx,T11]
OPER: movq $L1, %'d0 D:[T12] S:[]
MOVE: movq %'s0, %'d0 D:T0 S:T12
OPER: movq $L2, %'d0 D:[T13] S:[]
MOVE: movq %'s0, %'d0 D:T1 S:T13
OPER: movq $1, %'d0 D:[T14] S:[]
MOVE: movq %'s0, %'d0 D:T3 S:T14
OPER: pushq 's0 D:[] S:[T0]
OPER: pushq 's0 D:[] S:[T1]
OPER: call _stringCompare D:[rax,r10,r11] S:[]
OPER: addq $16, %'d0 D:[rsp] S:[rsp]
MOVE: movq %'s0, %'d0 D:T2 S:rax
OPER: movq $0, %'d0 D:[T15] S:[]
OPER: cmpq 's0, 's1 D:[] S:[T2,T15]
OPER: je 'j0 D:[] S:[] J:[L3,L4]
LABEL: L4
OPER: movq $0, %'d0 D:[T16] S:[]
MOVE: movq %'s0, %'d0 D:T3 S:T16
LABEL: L3
MOVE: movq %'s0, %'d0 D:rax S:T3
MOVE: movq %'s0, %'d0 D:rdi S:T4
MOVE: movq %'s0, %'d0 D:rsi S:T5
MOVE: movq %'s0, %'d0 D:rbx S:T6
MOVE: movq %'s0, %'d0 D:r12 S:T7
MOVE: movq %'s0, %'d0 D:r13 S:T8
MOVE: movq %'s0, %'d0 D:r14 S:T9
MOVE: movq %'s0, %'d0 D:r15 S:T10
OPER: jmp 'j0 D:[] S:[] J:[L5]
LABEL: L5
OPER:  D:[] S:[rax,rsp,rbp,rdi,rsi,rbx,r12,r13,r14,r15]
;;-END-FRAME-:
yes!!
